src/adapter/src/coord.rs

```diff
@@ -1885,11 +1894,22 @@ impl Coordinator {
         Arc::make_mut(&mut self.catalog)
     }
 
-    /// Obtain writeable Catalog and Controller references. This function is
-    /// needed to allow rust to have multiple mutable references on self at the
-    /// same time.
-    fn catalog_and_controller_mut(&mut self) -> (&mut Catalog, &mut mz_controller::Controller) {
-        (Arc::make_mut(&mut self.catalog), &mut self.controller)
+    /// Obtain writeable Catalog and Controller references. This function is needed to allow rust to
+    /// have multiple mutable references on self at the same time. An optional ConnectionId can be
+    /// passed to get a reference to an active ConnMeta.
+    fn catalog_and_controller_mut(
```

Comment:
I think another way to do this is to destructure the `Coordinator` struct, e.g.

```
let Coordinator {
  catalog,
  controller,
  active_conns,
  ..
} = &mut self;
```

I'm calling this out because calling `catalog_and_controller_mut(None)` maybe feels a bit awkward. It's not a huge deal but I could see this function growing and eventually becoming unwieldy

Reply:
This worked much better, thanks.